geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/collections/SizeableByteArrayList.java

```diff
@@ -31,6 +33,57 @@
       roundUpSize(getObjectHeaderSize() + 3 * getReferenceSize());
   private int memberOverhead;
 
+  /**
+   * @param o element to remove from the list
+   * @param count number of elements that match object o to remove from the list.
+   *        Count that is equal to 0 removes all matching elements from the list.
+   * @return list of indexes that were removed in order.
+   */
+  public List<Integer> removeObjectsStartingAtHead(Object o, int count) {
```

Comment:
We have slightly separate code paths that are ultimately all dependent on whether count is positive or negative. Since this is a very specific use case we're creating these methods for, I think it would be cleaner to have this method receive either a positive or negative value and then do the appropriate thing. Also, the returned list should always be in the same order (ascending or descending) regardless of the value of count. That would simplify a lot of subsequent code.